pyfarm/agent/http/core/client.py

```diff
@@ -304,35 +323,56 @@ def request(method, url, **kwargs):
         The class to use to unpack the internal response.  This is mainly
         used by the unittests but could be used elsewhere to add some
         custom behavior to the unpack process for the incoming response.
+
+    :raises NotImplementedError:
+        Raised whenever a request is made of this function that we can't
+        implement such as an invalid http scheme, request method or a problem
+        constructing data to an api.
     """
-    # check assumptions for arguments
-    assert method in ("HEAD", "GET", "POST", "PUT", "PATCH", "DELETE")
-    assert isinstance(url, STRING_TYPES) and url
+    assert isinstance(url, STRING_TYPES)
+
+    # Make sure it's a method we support.
+    if method not in HTTP_METHODS:
+        raise NotImplementedError(
+            "This function only supports these http "
+            "methods: %s" % ", ".join(HTTP_METHODS))
+
+    # We only support http[s]
+    parsed_url = urlparse(url)
+    if parsed_url.scheme not in HTTP_SCHEMES:
+      raise NotImplementedError("Only http or https is supported.")
```

Comment:
Why do we need to catch that here manually? If treq does not support that scheme, it will raise an exception eventually anyway.